libfeed/feed-util.h

```diff
@@ -0,0 +1,30 @@
+/* Copyright 2018 Endless Mobile, Inc.
+ *
+ * eos-discovery-feed is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 2.1 of the
+ * License, or (at your option) any later version.
+ *
+ * eos-discovery-feed is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with eos-discovery-feed.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <glib-object.h>
+#include <feed-base-card-store.h>
+
+G_BEGIN_DECLS
+
+void
+maybe_append_card_type_construct_prop (GObjectConstructParam         *params,
+                                       guint                          n_params,
+                                       EosDiscoveryFeedCardStoreType  type);
```

Comment:
I should probably rename this func - initially I had it as `append` since I thought that default prop values weren't actually passed down the constructor, but they are, so this function actually has the effect of changing the prop value from `EOS_DISCOVERY_FEED_CARD_STORE_TYPE_UNSET` to something more useful.